gui/dist_view/distviewmodel.h

```diff
@@ -90,6 +94,9 @@ public slots:
 	 * (between subImage, addImage)
 	 */
 	bool inbetween;
+
+	bool coloringEnabled = false;
+	cv::Mat3f* coloringResult = nullptr;
```

Comment:
call it coloring and use a cv::Mat3f, not pointer. you can test it with .empty(). cv::Mat types use memory sharing so there is no copy involved (only header) in passing them around. don't pass pointer here.